internal/regexpchecker/regexpchecker.go

```diff
@@ -535,7 +547,7 @@ func (v *regExpValidator) scanUnicodePropertyValueExpression(isCharacterCompleme
 	start := v.pos - 3
 
 	propertyNameOrValueStart := v.pos
-	v.scanIdentifier(v.charAtOffset(0))
+	v.scanWordCharacters(v.charAtOffset(0))
```

Comment:
I'm not sure if your original intention in relaxing the checks was to suppress the "Expected '}'" error when the `}` appears after a several exotic identifier characters. If so, you might want to revert that commit.